WordPress/src/test/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModelTest.kt

```diff
@@ -172,15 +181,27 @@ class ActivityLogTypeFilterViewModelTest : BaseUnitTest() {
     @Test
     fun `items are checked, when the user opens the screen with active activity type filter`() = test {
         val uiStates = init().uiStates
-        val initialSelection = listOf(1, 4)
+        val initialSelection = listOf("1", "4")
 
         startVM(initialSelection = initialSelection)
 
         assertThat((uiStates.last() as Content).items.filterIsInstance(ActivityType::class.java)
-                        .filter { it.checked }.map { it.id }
+                .filter { it.checked }.map { it.id }
         ).containsExactlyElementsOf(initialSelection)
     }
 
+    @Test
+    fun `date range passed to the api, when provided on view model start`() = test {
+        init().uiStates
+        val captor: KArgumentCaptor<FetchActivityTypesPayload> = argumentCaptor()
+
+        startVM(dateRange = Pair(1L, 2L))
+
+        verify(activityLogStore).fetchActivityTypes(captor.capture())
+        assertThat(captor.firstValue.after!!.time).isEqualTo(1L)
+        assertThat(captor.firstValue.before!!.time).isEqualTo(2L)
```

Comment:
Question (❓): Maybe I asked that before, but since I would have used `?` instead of `!!`, trying to make sure we can be as consistent as possible next time, what was your reasoning behind using `!!` instead of `?`?

Reply:
I feel like that overusing "?" just leads to hidden bugs which are harder to track down. I'm trying to follow the fail fast principle even in these cases. 

In this particular test it doesn't really matter, for two reasons
1. "time" is not nullable, but imagine that "time" would be a nullable field. If it failed with "null but 2L was expected", you might be trying to figure out why "time" was null. However, when you use "!!" you can see right away, that "before" was null not "time". 

2. These are the last two lines of the test, but imagine it was used in the middle of the method. 


In general, I think that "?" should be used for cases when
1. null value is expected and it is a valid value
2. working with legacy APIs

If the state is not expected, I think the app/test should crash as soon as possible (fail fast principle).

Reply:
Thanks for sharing again Jirka! 🤗 

I am still not sure, but I urge you to have this mindset on my future PRs and let me know when I do use `?` instead of `!!`. I would love to know more of how you think about it, since I am still unsure about it. For me atm, `tests` are test, they either pass or fail. Adding `crash` logic and thinking into test is another level that I might need to explore more. So far, I only had the tests `crash` when I was trying to verify the `exception`.

Again, thanks for sharing and please keep helping me in understanding this approach more. 🤗

Reply:
I think that using this approach in tests might be a bit too much. At the same time valid usage of `!!` is way more common in tests than in regular code. Therefore, I like to use the same approach even in tests so people get use to not overusing `?`.

> Adding crash logic and thinking into test is another level that I might need to explore more.

Actually, you are not adding crash logic, you are just not adding not safe-guarding against null.